rubin_sim/scheduler/basis_functions/basis_functions.py

```diff
@@ -1949,3 +1950,57 @@ def _calc_value(self, conditions, indx=None):
         reward_map[mask] = np.nan
 
         return reward_map
+
+
+class BalanceVisits(Base_basis_function):
+    """Balance visits across multiple surveys.
+
+    Parameters
+    ----------
+    nobs_reference : int
+        Expected number of observations across all interested surveys.
+    note_survey : str
+        Note value for the current survey.
+    note_interest : str
+        Substring with the name of interested surveys to be accounted.
+    nside : int
+        Healpix map resolution.
+
+    Notes
+    -----
+    This basis function is designed to balance the reward of a group of
+    surveys, such that the group get a reward boost based on the required
+    collective number of observations.
+
+    For example, if you have 3 surveys (e.g. SURVEY_A_REGION_1,
+    SURVEY_A_REGION_2, SURVEY_A_REGION_3), when one of them is observed once
+    (SURVEY_A_REGION_1) they all get a small reward boost proportianal to the
+    collective number of observations (`nobs_reference`). Further observations
+    of SURVEY_A_REGION_1 would now cause the other surveys to gain a reward
+    boost in relative to it.
+    """
+
+    def __init__(self, nobs_reference, note_survey, note_interest, nside=None):
+        super().__init__(nside=nside)
+
+        self.nobs_reference = nobs_reference
+
+        self.survey_features = {}
+        self.survey_features["N_obs_survey"] = features.N_obs_survey(note=note_survey)
+        self.survey_features["N_obs_survey_interest"] = features.N_obs_survey(
+            note=note_interest
+        )
+
+    def _calc_value(self, conditions, indx=None):
+
+        return (
+            1
+            + np.floor(
+                self.survey_features["N_obs_survey_interest"].feature
+                / self.nobs_reference
+            )
+        ) / (
+            self.survey_features["N_obs_survey"].feature
+            if self.survey_features["N_obs_survey"].feature > 0
+            else 1
+        )
```

Comment:
Somehow, this sounds very similar to the other basis function added - RewardNObsSequence - so I guess I don't understand how it's different or the same?

Reply:
So, I know they sound similar but they are actually pretty complementary. This basis function is for the cases when a single survey observation is comprised of multiple observations. Think Deep Drilling surveys. On DD this is not really an issue because their rewards is either -inf or +inf (or something like that). But the survey I use for AuxTel is a small variation of DD where the reward is actually a value. And we merge observations in different filters, same way we normally do with DD. But, because this survey is a value, it happens sometime that the value of another survey wins so we don't complete the observations. This basis function gives a boost in the reward once you did 1 observation and continues until to completes the total number of observations.

The other basis function os more on tying together observations of different surveys.